flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/SqlCommandParser.java

```diff
@@ -256,7 +256,7 @@ private static SqlCommandCall parseBySqlParser(Parser sqlParser, String stmt) {
 
         USE_CATALOG,
 
-        USE,
+        USE("USE\\s+(.*)", SINGLE_OPERAND),
```

Comment:
I agree this can fix the problem. However, this fallbacks to the old way which uses regex to parse the SQL. We should use the TableEnvironment to parse the common SQL. 

Therefore, a simple fix can be adding quotes for identifiers in `org.apache.flink.table.client.cli.CliClient#callUseDatabase`.

Reply:
I have revised it according to your opinion